client/packages_test.go

```diff
@@ -121,3 +121,104 @@ func (cs *clientSuite) TestClientPackages(c *check.C) {
 		},
 	})
 }
+
+func (cs *clientSuite) TestPackageIsInstalled(c *check.C) {
+	isInstalledTests := map[string]bool{
+		"": false,
+		client.StatusNotInstalled: false,
+		client.StatusInstalled:    true,
+		client.StatusActive:       true,
+	}
+
+	for status, result := range isInstalledTests {
+		pkg := client.Package{Status: status}
+		c.Assert(pkg.IsInstalled(), check.Equals, result)
+	}
+}
+
+func (cs *clientSuite) TestPackagesInstalled(c *check.C) {
+	packages := client.Packages{
+		client.Package{Name: "not installed", Status: client.StatusNotInstalled},
+		client.Package{Name: "installed", Status: client.StatusInstalled},
+		client.Package{Name: "removed", Status: client.StatusRemoved},
+	}
+
+	installed := packages.Installed()
+	c.Assert(installed, check.HasLen, 1)
+	c.Assert(installed[0].Name, check.Equals, "installed")
+}
+
+func (cs *clientSuite) TestPackageHasNameMatching(c *check.C) {
+	pkg := client.Package{Name: "my package"}
+
+	hasNameMatchingTests := map[string]bool{
+		"*":    false,
+		"":     true,
+		"pack": true,
+		"MY":   true,
+		"myp":  false,
+	}
+
+	for query, result := range hasNameMatchingTests {
+		c.Assert(pkg.HasNameMatching(query), check.Equals, result)
+	}
+}
+
+func (cs *clientSuite) TestPackagesNamesMatching(c *check.C) {
+	packages := client.Packages{
+		client.Package{Name: "first app"},
+		client.Package{Name: "second app"},
+		client.Package{Name: "third app"},
+	}
+
+	matching := packages.NamesMatching("second")
+	c.Assert(matching, check.HasLen, 1)
+	c.Assert(matching[0].Name, check.Equals, "second app")
+}
+
+func (cs *clientSuite) TestPackageHasTypeInSet(c *check.C) {
+	pkg := client.Package{Type: client.TypeFramework}
+
+	hasTypeInSetTest := []struct {
+		types  []string
+		result bool
+	}{
+		{[]string{}, false},
+		{[]string{client.TypeFramework}, true},
+		{[]string{client.TypeApp, client.TypeFramework}, true},
+		{[]string{client.TypeKernel}, false},
+	}
+
+	for _, tt := range hasTypeInSetTest {
+		c.Assert(pkg.HasTypeInSet(tt.types), check.Equals, tt.result)
+	}
+}
+
+func (cs *clientSuite) TestPackagesTypesInSet(c *check.C) {
+	packages := client.Packages{
+		client.Package{Name: "app", Type: client.TypeApp},
+		client.Package{Name: "framework", Type: client.TypeFramework},
+		client.Package{Name: "kernel", Type: client.TypeKernel},
+	}
+
+	matching := packages.TypesInSet([]string{client.TypeFramework})
+	c.Assert(matching, check.HasLen, 1)
+	c.Assert(matching[0].Name, check.Equals, "framework")
+}
+
+func (cs *clientSuite) TestPackagesSortByName(c *check.C) {
+	packages := client.Packages{
+		client.Package{Name: "c"},
+		client.Package{Name: "b"},
+		client.Package{Name: "a"},
+	}
+
+	sorted := packages.SortByName()
+
+	var sortedNames []string
```

Comment:
(nitpick) This could also be written as:
```
-       var sortedNames []string
-       for _, pkg := range sorted {
-               sortedNames = append(sortedNames, pkg.Name)
-       }
-
-       c.Assert(sortedNames, check.DeepEquals, []string{"a", "b", "c"})
+       c.Assert(sorted, check.DeepEquals, client.Packages{{Name: "a"}, {Name: "b"}, {Name: "c"}})
```
which is slightly shorter.